auslib/admin/views/permissions.py

```diff
@@ -95,3 +95,41 @@ def _delete(self, username, permission, changed_by, transaction):
         except ValueError as e:
             self.log.warning("Bad input: %s", e.args)
             return Response(status=400, response=e.args)
+
+
+class UserRolesView(AdminView):
+    """/users/:username/roles"""
+
+    def get(self, username):
+        roles = dbo.permissions.user_roles.getRoles(username)
+        if roles:
+            return jsonify({"roles": roles})
+        else:
+            return Response(status=404, response="No roles found for user")
+
+
+class UserRoleView(AdminView):
+    """/users/:username/roles/:role"""
+
+    @requirelogin
+    def _put(self, username, role, changed_by, transaction):
+        r = dbo.permissions.user_roles.select({"username": username, "role": role}, transaction=transaction)
+        if r:
+            return Response(status=200, response=json.dumps({"new_data_version": r[0]["data_version"]}))
+
+        dbo.permissions.user_roles.insert(changed_by, transaction, username=username, role=role)
+        return Response(status=201, response=json.dumps({"new_data_version": 1}))
+
+    @requirelogin
+    def _delete(self, username, role, changed_by, transaction):
+        if role not in dbo.permissions.user_roles.getRoles(username):
+            return Response(status=404)
+
+        form = DbEditableForm(request.args)
```

Comment:
Because username and role are part of the URL, there's no need to create a Form (they don't show up in request.args/data). You can see similar with Permissions, where "options" is the only thing that's part of the form. This code can't fire without them being set because of the routing!